src/forbidden/ad-test-signatures.txt

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+@defaultMessage use format with Locale
+java.lang.String#format(java.lang.String,java.lang.Object[])
+java.util.concurrent.ThreadLocalRandom
+
+java.security.MessageDigest#clone() @ use org.opensearch.common.hash.MessageDigests
+
+@defaultMessage Don't use MethodHandles in slow ways, don't be lenient in tests.
+java.lang.invoke.MethodHandle#invoke(java.lang.Object[])
+java.lang.invoke.MethodHandle#invokeWithArguments(java.lang.Object[])
+java.lang.invoke.MethodHandle#invokeWithArguments(java.util.List)
+java.lang.Math#random() @ Use one of the various randomization methods from LuceneTestCase or OpenSearchTestCase for reproducibility
```

Comment:
Why do we limit these? You said use MethodHandles. What alternatives do you suggest to use?

Reply:
I transferred over these checks and message from what is done in opensearch core. The alternative to using invoke is invokeExact which requires an exact type match. Using invokeWithArguments or invoke is also slower. I didn't give this too much thought as we aren't using MethodHandles at all. I can change this to remove invokeWithArgument which don't have as a direct replacement for different cases as invoke has a replacement in invokeExact and specify invokeExact alternative in warning message. Using invokeExact instead of invoke made sense to me but again we aren't using it at all and I personally haven't spent time using them other then investigating reason in opensearch core so I can remove all of these warnings for methodHandle

Reply:
You can keep them. Please specify alternative in warning message.